src/auth/KeychainStorage.ts

```diff
@@ -0,0 +1,59 @@
+import {Platform} from 'react-native'
+import * as Keychain from 'react-native-keychain'
+
+async function write(key: string, value: string) {
+  return Keychain.setGenericPassword(key, value, {
+    service: key,
+    accessControl: Keychain.ACCESS_CONTROL.BIOMETRY_ANY_OR_DEVICE_PASSCODE,
+    accessible: Keychain.ACCESSIBLE.WHEN_UNLOCKED_THIS_DEVICE_ONLY,
+    securityLevel: Keychain.SECURITY_LEVEL.SECURE_HARDWARE,
+  }).then((result) => {
+    if (result === false) Promise.reject('Unable to store secret')
+  })
+}
+
+async function read(key: string, authenticationPrompt: Keychain.Options['authenticationPrompt']) {
+  const credentials: false | Keychain.UserCredentials = await Keychain.getGenericPassword({
+    service: key,
+    authenticationPrompt,
+    accessControl: Keychain.ACCESS_CONTROL.BIOMETRY_CURRENT_SET_OR_DEVICE_PASSCODE,
+  })
+
+  if (!credentials) throw new Error('Failed to load secret')
+  return credentials.password
+}
+
+async function remove(key: string) {
+  return Keychain.resetGenericPassword({
+    service: key,
+  })
+}
+
+const KEYCHAIN_APP_AUTH_KEY = 'os-authentication'
+const initializeAppAuth = () => write(KEYCHAIN_APP_AUTH_KEY, KEYCHAIN_APP_AUTH_KEY) // value is irrelevant
+const appAuth = (authenticationPrompt: Keychain.Options['authenticationPrompt']) =>
```

Comment:
is the return data used for anything? if not, return void

Reply:
~appAuth~ -> `authenticate`

Reply:
`authenticate` for can mislead to bio authenticate. that's why is appAuth, any better suggestion?